src/movies.js

```diff
@@ -1,26 +1,164 @@
 // The `movies` array from the file `src/data.js`.
-console.log('movies: ', movies);
+//console.log('movies: ', movies);
+
+const movies = require("./data");
+
 
 
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllDirectors(arrayMovies) {
+let newArrayDirectors = arrayMovies.map(function (array) {
+return array.director;
+})
+return newArrayDirectors;
+}
+//console.log (getAllDirectors(movies));
+
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+
+function howManyMovies(arrayMovies) {
+  let numberOfDramaMoviesSpielberg = 0;
+  if(arrayMovies.length===0) return 0;
+    else {
+    let dramaMovies = arrayMovies.filter(array => {
+    if (array.director==="Steven Spielberg") {
+      for(let i=0; i<array.genre.length; i++) {
+            if (array.genre[i]==="Drama") {
+              numberOfDramaMoviesSpielberg++;
+              return array;
+            }
+      }
+    }
+    }
+    );
+  }
+  if (numberOfDramaMoviesSpielberg===0) return 0;
+  else return numberOfDramaMoviesSpielberg;
+}
+//console.log(howManyMovies(movies));
+
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+function scoresAverage(array) {
+  if(array.length===0) return 0;
+    else {
+      let numberOfScores=0;
+      const sumScores = array.reduce (function (accumulator, movie) {
+        if(movie.score) {
+          numberOfScores++;
+          return accumulator + movie.score;
+        } else return accumulator;
+      }, 0);
+      return (sumScores / numberOfScores).toFixed(2);
```

Comment:
Definition of `toFixed` method:
Returns a string representing a number in fixed-point notation.
you needed to convert it back to a number